lib/authz/permissions.go

```diff
@@ -1041,7 +1041,8 @@ func definitionForBuiltinRole(clusterName string, recConfig types.SessionRecordi
 						types.NewRule(types.KindOktaAssignment, services.RW()),
 						types.NewRule(types.KindProxy, services.RO()),
 						types.NewRule(types.KindClusterAuthPreference, services.RO()),
-						types.NewRule(types.KindRole, services.RO()),
+						types.NewRule(types.KindRole, services.RW()),
+						types.NewRule(types.KindAccessList, services.RW()),
```

Comment:
I have a problem with this fragment: I don't understand it. The documentation of this function says we construct an "appropriate role definition for a given builtin role". What is an "appropriate role definition"? What is its significance in the system? Does it mean that components of our system (nodes etc.) run assuming roles that are also available to users, or does it mean creating some sort of role definitions required by user to access a given system represented by a "system role"?